pkg/solidfire/types.go

```diff
@@ -542,3 +556,80 @@ type ListVolumeAccessGroupsResponse struct {
 		} `json:"volumeAccessGroups"`
 	} `json:"result"`
 }
+
+type ListVirtualVolumeTasksResponse struct {
+	ID     int `json:"id"`
+	Result struct {
+		Tasks []struct {
+			Cancelled            bool   `json:"cancelled"`
+			CloneVirtualVolumeID string `json:"cloneVirtualVolumeID"`
+			Operation            string `json:"operation"`
+			ParentMetadata       struct {
+				SFProfileID           string `json:"SFProfileId"`
+				SFgenerationID        string `json:"SFgenerationId"`
+				VMWContainerID        string `json:"VMW_ContainerId"`
+				VMWGosType            string `json:"VMW_GosType"`
+				VMWVVolName           string `json:"VMW_VVolName"`
+				VMWVVolNamespace      string `json:"VMW_VVolNamespace"`
+				VMWVVolType           string `json:"VMW_VVolType"`
+				VMWVMID               string `json:"VMW_VmID"`
+				VMWVvolAllocationType string `json:"VMW_VvolAllocationType"`
+				VMWVvolProfile        string `json:"VMW_VvolProfile"`
+			} `json:"parentMetadata"`
+			ParentTotalSize     int64  `json:"parentTotalSize"`
+			ParentUsedSize      int    `json:"parentUsedSize"`
+			Status              string `json:"status"`
+			VirtualVolumeHostID string `json:"virtualVolumeHostID"`
+			VirtualVolumeTaskID string `json:"virtualVolumeTaskID"`
+			VirtualvolumeID     string `json:"virtualvolumeID"`
+		} `json:"tasks"`
+	} `json:"result"`
+}
+
+type ListBulkVolumeJobsResponse struct {
+	ID     int64 `json:"id"`
+	Result struct {
+		BulkVolumeJobs []struct {
+			Attributes struct {
+				BlocksPerTransfer int64  `json:"blocksPerTransfer"`
+				FirstPendingLba   int64  `json:"firstPendingLba"`
+				NLbas             int64  `json:"nLbas"`
+				NextLba           int64  `json:"nextLba"`
+				PendingLbas       string `json:"pendingLbas"`
+				PercentComplete   int64  `json:"percentComplete"`
+				StartLba          int64  `json:"startLba"`
+			} `json:"attributes"`
+			BulkVolumeID    int64  `json:"bulkVolumeID"`
+			CreateTime      string `json:"createTime"`
+			ElapsedTime     int64  `json:"elapsedTime"`
+			Format          string `json:"format"`
+			Key             string `json:"key"`
+			PercentComplete int64  `json:"percentComplete"`
+			RemainingTime   int64  `json:"remainingTime"`
+			Script          string `json:"script"`
+			SnapshotID      int64  `json:"snapshotID"`
+			SrcVolumeID     int64  `json:"srcVolumeID"`
+			Status          string `json:"status"`
+			Type            string `json:"type"`
+		} `json:"bulkVolumeJobs"`
+	} `json:"result"`
+}
+
+type ListAsyncResultsResponse struct {
+	ID     int64 `json:"id"`
+	Result struct {
+		AsyncHandles []struct {
+			AsyncResultID int64  `json:"asyncResultID"`
+			Completed     bool   `json:"completed"`
+			CreateTime    string `json:"createTime"`
+			Data          struct {
```

Comment:
It won't affect anything yet, but the fields in `Data` can change depending on the value of `ResultType`.  I don't think we need actually implement all those since I'm not sure we'd ever use them but it'd be good to change this to a `map[string]any`

Reply:
Done